AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -458,15 +461,16 @@ am::am_Error_e CAmSocketHandler::addFDPoll(const int fd, const short event, IAmS
 /**
   * removes a filedescriptor from the poll loop
   * @param handle
+  * @param [close]
   * @return
   */
-am_Error_e CAmSocketHandler::removeFDPoll(const sh_pollHandle_t handle)
+am_Error_e CAmSocketHandler::removeFDPoll(const sh_pollHandle_t handle, const bool close)
```

Comment:
(Style)

Maybe you could pass in poll_states_e::CLOSE or poll_states_e::REMOVE as parameter instead?  Those are the only possible values here, right?

close = false, actually means "remove" - it's not obvious...

It's a common API style guideline to avoid boolean params.   There are exceptions to the rule as always.  But generally.   The point is that reading the code from the calling side, it's almost impossible to know what a boolean stands for, so you have to go and look inside the function to know... and that breaks the abstraction value of the function.

(see next comment on the function call)

Reply:
close = false means that the fd will be removed according to the old implementation.
close = true means that the fd will be removed and closed.
I would prefer to not use the internal states. They are not needed to export for applications.
I could only imaging the define two new parameters like:
#define RMV_ONLY false
#define RMV_N_CLOSE true

Another option would be to provide a new function:
CAmSocketHandler::removeAndCloseFDPoll()

Any other suggestion?

Reply:
>  I would prefer to not use the internal states. They are not needed to export for applications.

Fair point.  I didn't really check what is appropriate scope of the states.

Yeah, new #defines also feels a bit heavyweight for the problem.

OK, probably cleanest is the last option.  Two different functions that then call a third for the common implementation (which could be an inline if performance is a concern).

Since there's no real good option here, maybe just comment on the calling side.